### ISSUE TITLE
[Warp Drive] programmatic export of entire warp drive

### ISSUE THREAD

**Issue by ipsod** [warpdotdev/Warp#5356]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

a bash command, like `warp export_drive /path/to/backup`

### Is your feature request related to a problem? Please describe.

As part of my backup strategy, I'd like to be able to export the contents of my drive programmatically.

It's business-critical info, and I have automatic backups in place for such info, and don't know how to automate backups for Warp drive stuff.

### How important is this feature to you?

5 (very important)

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by dannyneira** [comment]
Hi @ipsod, Currently, there is no way to do this programmatically, only through the Command Palette, by searching for "Export All Warp Drive Objects". That being said since Warp drive is backed by GCP, there shouldn't be an issue with redundancy since that comes out of the box.

That being said, I could see an URI for exporting or some other method being helpful for local backups of your Warp Drive so thanks for this feature request! 

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by ipsod** [comment]
Oh, nice, at least there's a manual way.  Thanks for the quick reply.

**Comment by kmanc** [comment]
Hey @dannyneira, is the following still possible?

> ... through the Command Palette, by searching for "Export All Warp Drive > Objects".

I've tried searching for this in the command palette and I don't see this as an option. I don't necessarily need programmatic exports, but I don't see a manual way.


Thanks!

**Comment by dannyneira** [comment]
@kmanc yes it's still an option. just remove the ">" from your search, or just look for "export all". Please see more in our docs. https://docs.warp.dev/features/warp-drive#import-and-export

![Image](https://github.com/user-attachments/assets/1fa651dd-8739-47a2-9571-40f3d7df6ae5)

Closing as resolved since there is option for export / import to Warp Drive.

**Comment by kmanc** [comment]
Ahhh ok I think I misunderstood the original ask then. @dannyneira hate to be a bother, but if my intent was/is to export all blocks from a session into something (say a PDF for example), would that be a separate feature request? Happy to spin one up but want to make sure it's not just me not misusing the tool
